pennylane/templates/subroutines/temporary_and.py

```diff
@@ -0,0 +1,255 @@
+
+        .. math:: O = O_1 O_2 \dots O_n.
+
+
+        .. seealso:: :meth:`~.TemporaryAnd.decomposition`.
+
+        Args:
+            wires (Sequence[int]): the subsystem the gate acts on. The first two wires are the control wires and the
+                third one is the target wire.
+            control_values (bool or int or list[bool or int]): The value(s) the control wire(s)
+                should take. Integers other than 0 or 1 will be treated as ``int(bool(x))``.
+
+        Returns:
+            list[Operator]: decomposition into lower level operations
+
+        **Example:**
+
+        >>> qml.TemporaryAnd.compute_decomposition((0,1,2))
+        [H(2),
+        T(2),
+        CNOT(wires=[1, 2]),
+        Adjoint(T(2)),
+        CNOT(wires=[0, 2]),
+        T(2),
+        CNOT(wires=[1, 2]),
+        Adjoint(T(2)),
+        H(2),
+        Adjoint(S(2))]
+        """
+
+        list_decomp = []
+
+        if control_values[0] == 0:
+            list_decomp.append(qml.X(wires[0]))
+
+        if control_values[1] == 0:
+            list_decomp.append(qml.X(wires[1]))
+
+        list_decomp += [
+            qml.Hadamard(wires=wires[2]),
+            qml.T(wires=wires[2]),
+            qml.CNOT(wires=[wires[1], wires[2]]),
+            qml.adjoint(qml.T(wires=wires[2])),
+            qml.CNOT(wires=[wires[0], wires[2]]),
+            qml.T(wires=wires[2]),
+            qml.CNOT(wires=[wires[1], wires[2]]),
+            qml.adjoint(qml.T(wires=wires[2])),
+            qml.Hadamard(wires=wires[2]),
+            qml.adjoint(qml.S(wires=wires[2])),
+        ]
+
+        if control_values[0] == 0:
+            list_decomp.append(qml.X(wires[0]))
+
+        if control_values[1] == 0:
+            list_decomp.append(qml.X(wires[1]))
+
+        return list_decomp
+
+
+def _temporary_and_resources(control_values):
+    number_xs = 4 - 2 * sum(control_values)
+    return {
+        qml.X: number_xs,
+        qml.Hadamard: 2,
+        qml.CNOT: 3,
+        qml.T: 2,
+        qml.decomposition.adjoint_resource_rep(qml.T, {}): 2,
+        qml.decomposition.adjoint_resource_rep(qml.S, {}): 1,
+    }
+
+
+@register_resources(_temporary_and_resources)
+def _temporary_and(wires: WiresLike, control_values):
+    if control_values[0] == 0:
+        qml.X(wires[0])
+
+    if control_values[1] == 0:
+        qml.X(wires[1])
+
+    qml.Hadamard(wires=wires[2])
+    qml.T(wires=wires[2])
+    qml.CNOT(wires=[wires[1], wires[2]])
+    qml.adjoint(qml.T(wires=wires[2]))
+    qml.CNOT(wires=[wires[0], wires[2]])
+    qml.T(wires=wires[2])
+    qml.CNOT(wires=[wires[1], wires[2]])
+    qml.adjoint(qml.T(wires=wires[2]))
+    qml.Hadamard(wires=wires[2])
+    qml.adjoint(qml.S(wires=wires[2]))
+
+    if control_values[0] == 0:
+        qml.X(wires[0])
+
+    if control_values[1] == 0:
+        qml.X(wires[1])
+
+
+add_decomps(TemporaryAnd, _temporary_and)
+# TODO: add add_decomps("Adjoint(TemporaryAnd)", _adjoint_TemporaryAnd) when MCMs supported by the pipeline
```

Comment:
Is this tracked in Shortcut?

Reply:
I don't think so
@AntonNI8 , where should be include this?